assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -2267,111 +2266,40 @@ public String getXmlAssignment(Assignment assignment) {
-//                                int length = grade.length();
-//                                if (length > dec) {
-//                                    decimalGradePoint = grade.substring(0, grade.length() - dec) + decSeparator + grade.substring(grade.length() - dec);
-//                                }
-//                                else {
-//                                    String newGrade = "0".concat(decSeparator);
-//                                    for (int i = length; i < dec; i++) {
-//                                        newGrade = newGrade.concat("0");
-//                                    }
-//                                    decimalGradePoint = newGrade.concat(grade);
-//                                }
-//                            }
-//                            catch (NumberFormatException e) {
-//                                try {
-//                                    Float.parseFloat(grade);
-//                                    decimalGradePoint = grade;
-//                                }
-//                                catch (Exception e1) {
-//                                    return grade;
-//                                }
-//                            }
-//                            // get localized number format
-//                            NumberFormat nbFormat = FormattedText.getNumberFormat(dec,dec,false);
-//                            DecimalFormat dcformat = (DecimalFormat) nbFormat;
-//                            // show grade in localized number format
-//                            try {
-//                                Double dblGrade = dcformat.parse(decimalGradePoint).doubleValue();
-//                                decimalGradePoint = nbFormat.format(dblGrade);
-//                            }
-//                            catch (Exception e) {
-//                                return grade;
-//                            }
-//                            return decimalGradePoint;
-//                        }
-//                        else
-//                        {
-//                            return StringUtils.trimToEmpty(grade);
-//                        }
-//                    }
-//                }
-//            }
-//        }
-//        return null;
-//    }
+    @Override
+    public String getGradeForSubmitter(String submissionId, String submitter) {
+        if (StringUtils.isAnyBlank(submissionId, submitter)) return null;
+        return getGradeForSubmitter(assignmentRepository.findSubmission(submissionId), submitter);
+    }
 
     @Override
-    public String getGradeForUserInGradeBook(String assignmentId, String userId) {
-        if (StringUtils.isAnyBlank(assignmentId, userId)) return null;
+    public String getGradeForSubmitter(AssignmentSubmission submission, String submitter) {
+        if (submission == null || StringUtils.isBlank(submitter)) return null;
```

Comment:
Don't need the `submission == null` check here, `StringUtils.isBlank` is checking this for you.